### PR TITLE
Add CI pipeline

### PR DESCRIPTION
**Pull request by efritz**
Part of https://github.com/sourcegraph/sourcegraph/issues/25100.

### CODE REVIEW

File: .buildkite/terraform-validate.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+set -ex
+
+cd "$(dirname "${BASH_SOURCE[0]}")"/..
+
+MODULES=(
+  modules/docker-mirror
+  modules/executors
+  modules/networking
+)
```
Comment (eseliger) [inline .buildkite/terraform-validate.sh:11]: I think we also want to validate `.`

**Review by eseliger — APPROVED** [review]

File: .buildkite/terraform-validate.sh (hunk 1 above)
Comment (efritz) [reply .buildkite/terraform-validate.sh:11]: I don't think we can with modules

Comment (eseliger) [reply .buildkite/terraform-validate.sh:11]: But there's also terraform stuff in the root, is that not possible to lint at least?

Comment (efritz) [reply .buildkite/terraform-validate.sh:11]: I spoke too soon (then deleted it)

**Review by eseliger — APPROVED** [review]

File: modules/executors/main.tf [hunk 2]
```diff
@@ -10,7 +10,7 @@ resource "google_compute_instance_template" "executor-instance-template" {
   tags = ["${local.prefix}executor"]
 
   labels = {
-    "executor_tag" = var.executor_tag
+    "executor_tag" = var.instance_tag
```
[the remark this answers is not in the document]
Comment (efritz) [reply modules/executors/main.tf:13]: Ah yeah this was just to get symmetry in the main module so we don't need `executor_executor_tag` or one that doesn't really have a prefix.

**Pull request merged**